Content.Server/Spider/SpiderSystem.cs

```diff
@@ -0,0 +1,90 @@
+using System.Linq;
+using Content.Server.Actions;
+using Content.Server.Popups;
+using Content.Shared.Spider;
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.MobState.Components;
+using Content.Shared.Doors.Components;
+using Content.Shared.Physics;
+using Content.Shared.Maps;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Spider
+{
+    public sealed class SpiderSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popup = default!;
+        [Dependency] private readonly ActionsSystem _action = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly IPrototypeManager _proto = default!;
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SpiderWebObjectComponent, ComponentStartup>(OnWebStartup);
+            SubscribeLocalEvent<SpiderComponent, ComponentStartup>(OnSpiderStartup);
+            SubscribeLocalEvent<SpiderComponent, SpiderWebActionEvent>(OnSpawnNet);
+        }
+
+        private void OnSpiderStartup(EntityUid uid, SpiderComponent component, ComponentStartup args)
+        {
+            var netAction = new InstantAction(_proto.Index<InstantActionPrototype>(component.WebActionName));
+            _action.AddAction(uid, netAction, null);
+        }
+
+        private void OnWebStartup(EntityUid uid, SpiderWebObjectComponent component, ComponentStartup args)
+        {
+            _appearance.SetData(uid,SpiderWebVisuals.Variant, _robustRandom.Next(1, 3));
+        }
+
+        private void OnSpawnNet(EntityUid uid, SpiderComponent component, SpiderWebActionEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            var transform = Transform(uid);
+
+            if (!_mapManager.TryGetGrid(transform.GridUid, out var grid))
+                return;
+
+            args.Handled = true;
```

Comment:
it changes nothing in this case since we dont return until the end after this, but we normally set the handled arguments true at the end of the method